x-pack/plugins/lens/public/xy_visualization/to_expression.ts

```diff
@@ -570,3 +529,25 @@ const extendedYConfigToExpression = (yConfig: ExtendedYConfig, defaultColor?: st
     ],
   };
 };
+
+const axisExtentConfigToExpression = (
+  extent: AxisExtentConfig | undefined,
+  layers: ValidXYDataLayerConfig[]
+): Ast => {
+  const hasLine = layers.filter(({ seriesType }) => seriesType.includes('line')).length > 0;
```

Comment:
This is not the same check - the old code would only execute this check for the series currently mapped to the axis, your code is behaving as if all series are mapped to all axes. This causes a problem in such a configuration:
<img width="2327" alt="Screenshot 2022-04-27 at 13 01 19" src="https://user-images.githubusercontent.com/1508364/165504239-aa9584b7-38ca-4fbd-8514-cac1470b5206.png">

All series mapped to the left axis are line chart, but it ignores the custom bounds because there is a bar series on the right axis. I think we can't do this calculation during expression building time.

Reply:
Turned back the validation of extent to the chart component.
<img width="1435" alt="Screenshot 2022-04-27 at 15 37 06" src="https://user-images.githubusercontent.com/22456368/165519484-7ef94876-ec89-4c99-9a29-dedf2291e65c.png">
<img width="1435" alt="Screenshot 2022-04-27 at 15 37 15" src="https://user-images.githubusercontent.com/22456368/165519525-0e227ca5-9f18-4ed1-8235-14f41bdf952e.png">